examples/specialty_plots/ishikawa.py

```diff
@@ -0,0 +1,229 @@
+=================
+
+Ishikawa Diagrams, fishbone diagrams, herringbone diagrams, or cause-and-effect
+diagrams are useful for visualizing the effect to many cause relationships
+Source: https://en.wikipedia.org/wiki/Ishikawa_diagram
+
+"""
+import logging
+from typing import List
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+import matplotlib
+
+_log = logging.getLogger(__name__)
+
+
+def drawspines(data: dict, ax: plt.Axes, parentspine: None, alpha_ps: float = 60.0,
+               alpha_ss: float = 0,
+               primary_spines_rel_xpos: List[float] = [np.nan],
+               spine_color: str = "black", reclevel: int = 1,
+               max_recursion_level: int = 4, debug: bool = False):
+    """
+    Draw an Ishikawa spine with recursion
+
+    Parameters
+    ----------
+    data : dict
+        dictionary structure containing problem, causes...
+    ax : plt.Axes
+        Matplotlib current axes
+    parentspine : None
+        Parent spine object
+    alpha_ps : float, optional
+        First spine angle using during recursion. The default is 60.0.
+    alpha_ss : float, optional
+        Secondary spine angle using during recursion. The default is 0.
+    primary_spines_rel_xpos : List[float], optional
+        Relative X-position of primary spines, a list of float is accepted.
+        The default is [np.nan].
+    spine_color : str, optional
+        Spine color. The default is "black".
+    reclevel : int, optional
+        Current recursion level. The default is 1.
+    max_recursion_level : int, optional
+        Maximum recursion level set. The default is 4.
+    debug : bool, optional
+        Print debug variables. The default is False.
+
+    Raises
+    ------
+    AttributeError
+        Maximum recursion level reached or passed a bad parentspine object format.
+
+    Returns
+    -------
+    None.
+
+    """
+    # stop recursion if maximum level reached
+    if reclevel > max_recursion_level:
+        raise AttributeError('Max Recursion Level Reached')
+
+    if isinstance(data, dict):
+        # switch to correct angle depending on recursion level
+        if reclevel % 2 != 0:
+            alpha = alpha_ps
+        else:
+            alpha = alpha_ss
+
+        if isinstance(parentspine, matplotlib.lines.Line2D):
+            # calculate parent data
+            ([xpb, xpe], [ypb, ype]) = parentspine.get_data()
+        elif isinstance(parentspine, matplotlib.text.Annotation):
+            xpb, ypb = parentspine.xy
+            xpe, ype = parentspine._x, parentspine._y  # parentspine._arrow_relpos
+        else:
+            raise AttributeError('Wrong Spine Graphical Element')
+
+        plen = np.sqrt((xpe - xpb) ** 2 + (ype - ypb) ** 2)
+        palpha = np.round(np.degrees(np.arctan2(ype - ypb, xpe - xpb)))
+
+        # calculate spine spacing
+        # calculate couple pairs, at least 1 pair to start at middle branch
+        pairs = len(list(data.keys())) // 2 + len(list(data.keys())) % 2 + 1
```

Comment:
If I've understood correctly:
```suggestion
        pairs = np.ceil(len(data) / 2) + 1
```